cartridge/feedback.lua

```diff
@@ -1,18 +1,68 @@
+local fio = require('fio')
 local log   = require('log')
 local json  = require('json')
 local fiber = require('fiber')
 local http  = require('http.client')
 
 local PREFIX = "feedback_daemon"
 
+local function parse_lua_file(name)
+    local state = {}
+    local eval = loadfile(name, 't', state)
+    pcall(eval)
+
+    return state
+end
+
+local function parse_manifest(filename)
+    local manifest = parse_lua_file(filename)
+    local deps = {}
+
+    for module, versions in pairs(manifest.dependencies or {}) do
+        for version, _ in pairs(versions) do
+            if deps[module] ~= nil then
+                break -- Decide how to deal with duplicates
+            end
+            deps[module] = version
+        end
+    end
+
+    return deps
+end
+
+local parsed_manifest = false
+
+local function fill_in_rocks(feedback)
+    feedback.rocks = feedback.rocks or {}
+
+    -- tarantool 1.10 doesn't support package.searchroot
+    -- so we are basically guessing by picking directory
+    -- of the init script. It is a default scenario
+    local app_dir = package.searchroot and package.searchroot() or fio.abspath(fio.dirname(arg[0]))
+    local manifest_path = fio.pathjoin(app_dir, '.rocks/share/tarantool/rocks/manifest')
+
+    if not parsed_manifest then
+        if fio.path.is_file(manifest_path) then
+            parsed_manifest = parse_manifest(manifest_path)
+        end
+    end
+
+    local rocks = parsed_manifest or {}
```

Comment:
Variables and functions naming should be revised all across the PR. How could you assign `parsed_manifest` to `rocks`? Does it hold rocks of manifest?

Reply:
`or {}` seems to be unnecessary. `parsed_manifest` can't be nil afaics.

Reply:
Agree. Thanks for pointing it out.

I've removed confusing `parsed_manifest` variable, and put everything under cached branch `if not rocks then ... end`

Reply:
> or {} seems to be unnecessary. parsed_manifest can't be nil afaics.

If manifest happen to be not a file, rocks may still be nil